src/query/src/dist_plan/merge_scan.rs

```diff
@@ -149,21 +150,28 @@ impl MergeScanExec {
         })
     }
 
-    pub fn to_stream(&self, _context: Arc<TaskContext>) -> Result<SendableRecordBatchStream> {
+    pub fn to_stream(&self, context: Arc<TaskContext>) -> Result<SendableRecordBatchStream> {
         let substrait_plan = self.substrait_plan.to_vec();
         let regions = self.regions.clone();
         let region_query_handler = self.region_query_handler.clone();
         let metric = MergeScanMetric::new(&self.metric);
         let schema = Self::arrow_schema_to_schema(self.schema())?;
 
+        let dbname = context.task_id().unwrap_or_default();
```

Comment:
The `task_id` in this line is confusing

Reply:
task_id is a datafusion concept and here it's been used to transfer dbname. I think it's better to wrap it with a encode/decode function to encode/decode dbname from task id .